### ISSUE TITLE
SURBL monitoring when facebook.com exists in blacklist 

### ISSUE THREAD

**Issue by wheldom01** [vstakhov/rspamd#1663]
### Classification (Please choose *one* option):

* [] Crash/Hang/Data loss
* [ ] WebUI/Usability
* [ ] Serious bug
* [x ] Other bug
* [ ] Feature
* [ ] Enhancement

### Reproducibility (Please choose *one* option):

* [X] Always
* [ ] Sometimes
* [ ] Rarely
* [ ] Unable
* [ ] I didn’t try
* [ ] Not applicable

### Rspamd version:
1.5.9

### Operation system, CPU, memory and environment:
Debian Jessie

### Description (Please provide a descriptive summary of the issue):
When a surbl contains the surbl monitoring address of facebook.com you get error messages in the log:

monitored; rspamd_monitored_dns_cb: DNS reply returned 'no error' for hostkarma.junkemailfilter.com while 'no records with this name' was expected

### Compile errors (if any):
n/a

### Steps to reproduce:
Add the following to local.d/rbl.conf and restart
rules {
    "HOSTKAMA_URIBL" {
        suffix = "hostkarma.junkemailfilter.com";
        noip = true;
        enabled = false;
        ips = {
                URIBL_HOSTKAMA_WHITE = "127.0.0.1";
                URIBL_HOSTKAMA_BLACK = "127.0.0.2";
                URIBL_HOSTKAMA_YELLOW = "127.0.0.3";
                URIBL_HOSTKAMA_BROWN = "127.0.0.4";
                URIBL_HOSTKAMA_NOBLACK = "127.0.0.5";
                URIBL_HOSTKAMA_24_48H = "127.0.2.1";
                URIBL_HOSTKAMA_LAST_10D = "127.0.2.2";
                URIBL_HOSTKAMA_OLDER_10D = "127.0.2.3";
        }
    }
}

### Expected results:
No error message in log

### Actual results:
monitored; rspamd_monitored_dns_cb: DNS reply returned 'no error' for hostkarma.junkemailfilter.com while 'no records with this name' was expected

### Configuration:
See above

### Additional information:
Suggest that the monitoring ip for rbls be changed from facebook.com to com

**Comment by vstakhov** [comment]
You are right: it should check for another address: 

~~~
Domain-name-based DNSxLs MUST contain an entry for the [RFC2606]
   reserved domain name "TEST" and MUST NOT contain an entry for the
   reserved domain name "INVALID".
~~~

However, I was concerning `facebook.com` as an example of **SANE** uribl behaviour. Perhaps, we can add a sort of hack for such RBLs or URIBLs to turn off monitoring or to switch it to some other domain.

**Comment by wheldom01** [comment]
Maybe a test that does a combination as described below??

    The combination of a test address that MUST exist and an address that
    MUST NOT exist allows a client system to check that a domain still
    contains DNSxL data, and to defend against DNSxLs that deliberately
    or by accident install a wildcard that returns an A record for all
    queries.  DNSxL clients SHOULD periodically check appropriate test
    entries to ensure that the DNSxLs they are using are still operating.

**Comment by vstakhov** [comment]
This combination test will be much more problematic and breaking. However, I've added the similar workaround.

**Comment by wheldom01** [comment]
Thank you

**Comment by fatalbanana** [comment]
Should be fixed in 1.6.0+. @wheldom01 - could you confirm?

**Comment by wheldom01** [comment]
I can confirm that this issue is indeed fixed. Been running in production for serveral days now thanks.
